applications/StructuralMechanicsApplication/custom_constitutive/linear_plane_strain.cpp

```diff
@@ -67,41 +67,35 @@ void  LinearPlaneStrain::CalculateMaterialResponsePK2 (Parameters& rValues)
     const double& NU    = MaterialProperties[POISSON_RATIO];
 
     //NOTE: SINCE THE ELEMENT IS IN SMALL STRAINS WE CAN USE ANY STRAIN MEASURE. HERE EMPLOYING THE CAUCHY_GREEN
-    if(Options.IsNot( ConstitutiveLaw::COMPUTE_STRAIN )) //large strains
+    if(Options.Is( ConstitutiveLaw::COMPUTE_STRAIN ))
     {
-
-        //1.-Compute total deformation gradient
-        const Matrix& F = rValues.GetDeformationGradientF();
-        
-        Matrix Etensor = prod(trans(F),F);
-        Etensor -= IdentityMatrix(2,2);
-        Etensor *= 0.5;
-        
-        noalias(StrainVector) = MathUtils<double>::StrainTensorToVector(Etensor);
+        CalculateStrain(rValues, StrainVector);
     }
 
+    if( Options.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR ) )
+    {
+        Matrix& ConstitutiveMatrix = rValues.GetConstitutiveMatrix();
+        CalculateElasticMatrix( ConstitutiveMatrix, E, NU );
+    }
+    
     if( Options.Is( ConstitutiveLaw::COMPUTE_STRESS ) )
     {
-        if( Options.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR ) ){
-
-          Matrix& ConstitutiveMatrix            = rValues.GetConstitutiveMatrix();
-          CalculateElasticMatrix( ConstitutiveMatrix, E, NU );
+        if (rValues.GetDeformationGradientF().size1() > 0) 
```

Comment:
Because like the law is linear it requires necessarily the Strain to compute the Stress, so if you set the Strain and you don't want to recompute the Strain you need some kind of check. Like there is not in the CL any function to check if the deformation gradient or whatever is defined I used the size of F. But it requires to define an empty F. We can solve this adding the function IsDefinedDeformationGradient or whatever to the constitutive_law.h But this is  more profound change